addons/website_version/models/experiment.py

```diff
@@ -0,0 +1,300 @@
+# -*- coding: utf-8 -*-
+from datetime import datetime, timedelta
+import logging
+from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT
+from openerp.exceptions import Warning
+from openerp import models, fields, exceptions, SUPERUSER_ID, api
+from openerp.tools.translate import _
+import simplejson
+
+_logger = logging.getLogger(__name__)
+
+
+class Experiment_version(models.Model):
+
+    _name = "website_version.experiment_version"
+
+    version_id = fields.Many2one('website_version.version', string="version_id", required=True, ondelete='cascade')
+    experiment_id = fields.Many2one('website_version.experiment', string="Experiment_id", required=True, ondelete='cascade')
+    frequency = fields.Selection([('10', 'Less'), ('50', 'Medium'), ('80', 'More')], string='Frequency', default='50')
+    google_index = fields.Integer(string='Google index')
+
+
+#This model allow to define the Google analytics goals in a experiment
+
+
+class Goals(models.Model):
+
+    _name = "website_version.goals"
+
+    name = fields.Char(string="Name", required=True)
+    google_ref = fields.Char(string="Reference Google", required=True)
+
+
+class Experiment(models.Model):
+
+    _name = "website_version.experiment"
+    _inherit = ['mail.thread']
+    _order = 'sequence'
+
+    @api.multi
+    def _check_view(self):
+        #No overlap for running experiments
+        for exp in self:
+            if exp.state == 'running':
+                for exp_ver in exp.experiment_version_ids:
+                    if exp_ver.search([('version_id.view_ids.key', 'in', [v.key for v in exp_ver.version_id.view_ids]), ('experiment_id', '!=', exp_ver.experiment_id.id), ('experiment_id.website_id', '=', exp_ver.experiment_id.website_id.id), ('experiment_id.state', '=', 'running')]):
+                        return False
+        return True
+
+    @api.multi
+    def _check_website(self):
+        for exp in self:
```

Comment:
return self.env['website_version.experiment.version'].search_count(
  [
    ('ids', 'in', self.mapped('experiment_version_ids').ids),
    ('version_id.website_id', '=', self.mapped('website_id').ids)
 ]) > 0 ?